library/alloc/src/string.rs

```diff
@@ -1849,6 +1849,34 @@ impl String {
         let slice = self.vec.into_boxed_slice();
         unsafe { from_boxed_utf8_unchecked(slice) }
     }
+
+    /// Consumes and leaks the `String`, returning a mutable reference to the contents,
+    /// `&'a mut str`.
+    ///
```

Comment:
It'd be good to add the same note as we have on `Vec::leak()`, that this function also leaks any extra capacity:
```suggestion
    ///
    /// This method does not reallocate or shrink the `String`,
    /// so the leaked allocation may include unused capacity that is not part
    /// of the returned slice.
```